### PR TITLE
fix totalDebt metric

### LINKED ISSUE #7751: totalDebt outdated after adjusting vault
## Describe the bug
On the latest master, testing on a local chain, looks like totalDebt is not updating after I created a vault with 1000 IST of debt and adjusted it down to 800 IST:

image

## To Reproduce
1. `cd packages/inter-protocol && scripts/start-local-chain.sh`
2. Go to dapp-inter and create a vault
3. Note the "IST Outstanding (Vaults)" updates to non-zero amount
4. Adjust the vault to lower the debt
5. Observe the "IST Outstanding (Vaults)" doesn't update
6. Adjust the vault again, see it update to the previous value after the prior adjustment

Tried reloading the page and double-checking with `agd query` to rule out the possibility of a UI bug

## Expected behavior
It updates at step 5, and at step 6 it shows the most up-to-date value

## Platform Environment
agoric-sdk @ f078d4221812788c19ac3df9febe5f7c1b11cb48

### PR DESCRIPTION
**Pull request by turadg**
closes: #7751

## Description

The `totalDebt` metric wasn't being updated when a vault was adjusted, in the `handleBalanceChange` callback. This is because it was handled elsewhere, but those mutations didn't cover all cases. To make it mutate in `handleBalanceChange` required shrinking the scope of the other two functions it had been mutated.

This also does a little code cleanup along the way.

### Testing Considerations

There are some small fudge numbers in the tests to make them pass. I didn't see clearly how to calculate where they come from.

### CODE REVIEW

**Comment by samsiegart** [comment]
Following the repro steps in https://github.com/Agoric/agoric-sdk/issues/7751 and I see the expected behavior now so LGTM. Leaving approval to @Chris-Hibbert

**Review by samsiegart — COMMENTED** [review]
LGTM

File: packages/inter-protocol/src/vaultFactory/vaultManager.js [hunk 1]
```diff
@@ -1040,11 +1038,15 @@ export const prepareVaultManagerKit = (
...
+          // total += vault's delta (post — pre)
           state.totalCollateral = AmountMath.subtract(
             AmountMath.add(state.totalCollateral, vault.getCollateralAmount()),
             oldCollateral,
           );
+          state.totalDebt = AmountMath.subtract(
+            AmountMath.add(state.totalDebt, vault.getCurrentDebt()),
+            oldDebtNormalized,
+          );
           // debt accounting managed through minting and burning
```
Comment (turadg) [inline packages/inter-protocol/src/vaultFactory/vaultManager.js]: this comment is no longer applicable. minting and burning were adjusting, but not publishing. so we do it here to cover both.

File: packages/inter-protocol/test/vaultFactory/test-vaultFactory.js [hunk 2]
```diff
@@ -1931,6 +1931,15 @@ test('manager notifiers, with snapshot', async t => {
     totalCollateral: { value: totalCollateral },
   });
 
+  // XXX fudgy numbers
+  totalCollateral += 2n;
+  totalDebt += 429n;
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/test/vaultFactory/test-vaultFactory.js]: These  are suspicious. If the point is to get the accounting right, we need to clarify.

**Review by Chris-Hibbert — DISMISSED** [review]

File: packages/inter-protocol/test/vaultFactory/test-vaultFactory.js (hunk 2 above)
Comment (turadg) [reply packages/inter-protocol/test/vaultFactory/test-vaultFactory.js]: Good call. Hunted down where the values come from and force pushed the small changes to these lines.

**Review by Chris-Hibbert — APPROVED** [review]
thanks for sleuthing it out.

**Pull request merged**